otter/test/models/test_cass_models.py

```diff
@@ -2274,6 +2092,185 @@ def not_empty_error(lockpath, recursive):
+    """
+    Tests for :func:`view_manifest`
+    """
+
+    def setUp(self):
+        """
+        Mock verified view
+        """
+        super(ViewManifestTests, self).setUp()
+        self.verified_view = patch(self, 'otter.models.cass.verified_view')
+        self.vv_return = {
+            'tenantId': self.tenant_id,
+            "groupId": self.group_id,
+            'id': "12345678g",
+            'group_config': serialize_json_data(self.config, 1.0),
+            'launch_config': serialize_json_data(self.launch_config, 1.0),
+            'active': '{"A":"R"}',
+            'pending': '{"P":"R"}',
+            'groupTouched': '2014-01-01T00:00:05Z.1234',
+            'policyTouched': '{"PT":"R"}',
+            'paused': False,
+            'desired': 0,
+            'created_at': 23
+        }
+        self.manifest = {
+            'groupConfiguration': self.config,
+            'launchConfiguration': self.launch_config,
+            'id': "12345678g",
+            'state': GroupState(
+                self.tenant_id,
+                self.group_id,
+                'a', {'A': 'R'},
+                {'P': 'R'}, '2014-01-01T00:00:05Z.1234',
+                {'PT': 'R'}, False)
+        }
+
+    def test_success(self):
+        """
+        When viewing the manifest, if the group exists a dictionary with the
+        config, launch config, and scaling policies is returned.
+        """
+        self.verified_view.return_value = defer.succeed(self.vv_return)
+        self.group._naive_list_policies = mock.Mock(
+            return_value=defer.succeed([]))
+
+        self.manifest['scalingPolicies'] = []
+        self.assertEqual(
+            self.validate_view_manifest_return_value(), self.manifest)
+
+        self.group._naive_list_policies.assert_called_once_with()
+
+        view_cql = (
+            'SELECT "tenantId", "groupId", group_config, launch_config, '
+            'active, pending, "groupTouched", "policyTouched", paused, '
+            'desired, created_at, status, deleting '
+            'FROM scaling_group '
+            'WHERE "tenantId" = :tenantId '
+            'AND "groupId" = :groupId')
+        del_cql = ('DELETE FROM scaling_group '
+                   'WHERE "tenantId" = :tenantId AND "groupId" = :groupId')
+        exp_data = {'tenantId': self.tenant_id, 'groupId': self.group_id}
+        self.verified_view.assert_called_once_with(
+            self.connection, view_cql, del_cql,
+            exp_data, ConsistencyLevel.QUORUM,
+            matches(IsInstance(NoSuchScalingGroupError)),
+            self.mock_log, get_deleting=False)
+
+    def test_with_webhooks(self):
+        """
+        Viewing manifest with_webhooks=True returns webhooks inside policies
+        that matches the `model_schemas.manifest`
+        """
+        self.verified_view.return_value = defer.succeed(self.vv_return)
+
+        # Getting policies
+        policies = [group_examples.policy()[i] for i in range(3)]
```

Comment:
Non-blocking - could just get the list once, and pick the first 3 elements.  :)  `group_examples.policy()[:3]`?